### PR TITLE
feat: refactor metainfo to nuxt hooks and fix seo 🛠️ 

### LINKED ISSUE #1700: Refactor vue-meta to the Nuxt approach
### Is your feature request related to a problem?

> why we even use `metaInfo()` (from vue-meta) at all? Nuxt uses vue-meta under the hood and provides much more powerful hooks via head() and meta. May be worth thinking about refactoring it all together. https://nuxtjs.org/docs/features/meta-tags-seo/

ht/ @kkukelka 

### Describe the solution you would like

- Use nuxt seo principles and use more reusable pieces 
- something like /utils/seo

### Describe alternatives you have considered

- 

### Screeenshots

-

### PR DESCRIPTION
**Pull request by kkukelka**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇  _ Do a quick check before the merge. 

### PR type
- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

Donkey Gang
Screenshot 2022-01-09 at 14 31 57

### Before submitting Pull Request, please make sure:
- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main-nuxt** and I've no conflicts
- [x] I've tried respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted screenshot of demonstrated change in this PR

### Optional
- [x] I've tested it at </rmrk/collection/0a8ce195286c168f19-DONKEY>
- [ ] I've tested PR on mobile and everything seems works
- [ ] I found edge cases

### What's new?
- [x] PR closes #1700 AND #1691
- [x] refactor `metaInfo()` with nuxt `head()` hook, fix all seo stuff and image links

### Had issue bounty label ?
- [x] Fill up your KSM address: [Payout](https://kodadot.xyz/transfer/?target=EqdyzrzVmeHwMdMwvPeCMnNdbuQDbD3YrjY93xq9Ln3jUGW)

### Community participation
- [x] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

### Screenshot
- [x] My fix has changed **something** on UI, a screenshot is best to understand changes for others.

### CODE REVIEW

File: pages/contact.vue [hunk 1]
```diff
@@ -5,13 +5,25 @@
...
 @Component({
   components: {
     Contact
+  },
+  head() {
+    const title = 'KodaDot cares about environmental impact'
+    const metaData = {
+      title: title,
+      url: `${this.$config.baseUrl}/contact`,
+      image: `${this.$config.baseUrl}/kodadot_carbonless.jpg`,
```
Comment (roiLeo) [inline pages/contact.vue:19]: old image here

File: utils/createSiteMeta.ts [hunk 2]
```diff
@@ -0,0 +1,60 @@
+const type = 'website'
+const url = 'https://nft.kodadot.xyz'
+const title = 'KodaDot - Kusama NFT Market Explorer'
+const description = 'Creating Carbonless NFTs on Kusama'
+const image = 'https://nft.kodadot.xyz/kodadot_card_root.png'
```
Comment (roiLeo) [inline utils/createSiteMeta.ts:5]: the fullpath are default values?

**Review by roiLeo — APPROVED** [review]
Woah! Big changes 🥇 
Screenshot 2022-01-10 at 08-23-20 Card Validator Twitter Developers

Screenshot 2022-01-10 at 08-28-31 Card Validator Twitter Developers

Any idea what was causing the issue of having readme content in meta description? Was it vueMeta config?

You solution is a good one, wouldn't it be better to import createSiteMeta as a plugin?

File: utils/createSiteMeta.ts (hunk 2 above)
Comment (kkukelka) [reply utils/createSiteMeta.ts:5]: yes but you actually made a good point, i think i will rewrite this function as a plugin -> then i will additionally have access to context & $config and can write something like `const url = this.$config.baseUrl` as default values 😸

**Comment by kkukelka** [comment]
@roiLeo 
Yes, the issue was partly because of vue-meta. Nuxt already uses it under the hood and has some pretty strict rules regarding how you are allowed to overwrite the default values we specified in nuxt.config: only via `head()` in page-components! So the metaInfo stuff we did in the individual components basically did nothing 😅 

but now we do it best-practice 👍

File: components/landing/Contact.vue [hunk 3]
```diff
@@ -17,24 +17,7 @@
...
-        { property: 'og:description', content: 'Creating Carbonless NFTs on Kusama'},
-        { property: 'twitter:title', content: 'KodaDOT cares about environmental impact'},
-        { property: 'twitter:description', content: 'Creating Carbonless NFTs on Kusama'},
-        { property: 'twitter:image', content: 'https://nft.kodadot.xyz/kodadot_carbonless.jpg'},
-      ]
-    }
-  },
-  components: {}
-})
+@Component<Contact>({})
```
Comment (vikiival) [inline components/landing/Contact.vue:20]:
```suggestion
@Component<Contact>
```

File: pages/rmrk/detail/_id.vue [hunk 4]
```diff
@@ -5,14 +5,45 @@
...
   name: 'GalleryItemPage',
   components: {
     GalleryItem
+  },
+  head() {
+    const title = this.currentlyViewedItem.title
+    const metaData = {
+      title: title,
+      description: this.currentlyViewedItem.description,
+      url: `https://nft.kodadot.xyz/${this.$route.path}`,
```
Comment (vikiival) [inline pages/rmrk/detail/_id.vue:21]: also old url?

File: pages/rmrk/detail/_id.vue [hunk 5]
```diff
@@ -5,14 +5,45 @@
...
+  }
+
+  get currentPrice(): string {
+    // need to remove whitespaces here since formatBalance return 'xxx KSM'
+    const price = this.currentlyViewedItem.price !== '0' ? formatBalance(this.currentlyViewedItem.price, 12, 'KSM').replaceAll(' ', '') : ''
+    return price
+  }
+
+  get image(): string {
+    return `https://og-image-green-seven.vercel.app/${encodeURIComponent(this.currentlyViewedItem.name)}.png?price=${this.currentPrice}&image=${this.currentlyViewedItem.image}&mime=${this.currentlyViewedItem.mimeType}`
```
Comment (vikiival) [inline pages/rmrk/detail/_id.vue:46]: I would move this to utils

File: pages/westmint/u/_id.vue [hunk 6]
```diff
@@ -155,50 +156,17 @@ const eq = (tab: string) => (el: string) => tab === el
 
 @Component<Profile>({
   components,
-  metaInfo() {
+  head() {
+    const title = 'NFT Artist Profile on KodaDot'
+    const metaData = {
+      title: title,
```
Comment (vikiival) [inline pages/westmint/u/_id.vue:162]:
```suggestion
      title,
```

File: utils/createSiteMeta.ts [hunk 7]
```diff
@@ -0,0 +1,60 @@
...
+
+export default (meta) => {
```
Comment (vikiival) [inline utils/createSiteMeta.ts:7]: I would make the function with a name.

also we can make return type like `SeoMeta[]`

```ts
type SeoMeta = {
  hid: string,
  name: string,
  constent: string
}
```

**Review by vikiival — COMMENTED** [review]
Otherwise good

File: components/landing/Contact.vue (hunk 3 above)
Comment (kkukelka) [reply components/landing/Contact.vue:20]: this doesn't work. `@Component` need at least 1 argument for its constructor, even if its empty

File: utils/createSiteMeta.ts (hunk 7 above)
Comment (kkukelka) [reply utils/createSiteMeta.ts]: i believe `MetaInfo['meta']` to be the correct return type.

File: pages/contact.vue (hunk 1 above)
Comment (kkukelka) [reply pages/contact.vue]: changed to `k_card_mint`

File: pages/rmrk/detail/_id.vue (hunk 4 above)
Comment (kkukelka) [reply pages/rmrk/detail/_id.vue]: reworked completely, url now via baseurl in plugin

File: pages/rmrk/detail/_id.vue (hunk 5 above)
Comment (kkukelka) [reply pages/rmrk/detail/_id.vue]: now in `seoImageGenerator.ts` util

**Comment by yangwao** [comment]
are we ready to fly?

/giphy my first flight

**Comment by kkukelka** [comment]
> are we ready to fly?
> 
> /giphy my first flight

gtg from my side.

**Comment by yangwao** [comment]
yolo

Perfect, I’ve sent the payout https://kusama.subscan.io/extrinsic/0xdfe00088121789889406710c2efd68964f775f594bf723cee9eb10072863e2ba

Let’s grab another issue and get rewarded! 
https://github.com/kodadot/nft-gallery/issues/

**Pull request merged**